src/include/txn.i

```diff
@@ -896,6 +898,34 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_UPDATE **updp)
 			skipped_birthmark = true;
 	}
 
+	/*
+	 * If there is a related lookaside entry, find it and try matching the
+	 * in-memory record. The following conditions will be refined in the
+	 * future once we make the change to stop initializing a page using
+	 * lookaside.
+	 */
+	if (upd != NULL && upd->txnid != 0 && cbt->ref->page_las != NULL &&
+	    __wt_find_lookaside_upd(session, cbt, &tmp_upd, true) == 0) {
+		/* It is okay to locate an older update in lookaside */
+		WT_ASSERT(session, upd->start_ts >= tmp_upd->start_ts);
+
+		if (upd->start_ts == tmp_upd->start_ts) {
+			/*
+			 * If we located the correct record, the contents should
+			 * be the same and we can use the record from lookaside.
+			 */
+			WT_ASSERT(session, (upd->size == tmp_upd->size &&
+			    memcmp(upd->data, tmp_upd->data, upd->size) == 0));
+			upd = tmp_upd;
+			/*
+			 * Mark this update as external and to be discarded
+			 * when not needed.
+			 */
+			upd->ext = true;
+		} else
+			__wt_free_update_list(session, tmp_upd);
```

Comment:
If `start_ts is passed into `__wt_find_lookaside_upd()`, then `upd.ext` can be set there, and there's no need to free the update list (in short, this work gets simpler)?